src/model.jl

```diff
@@ -58,18 +58,54 @@ Given a Case object and a DemandFlexibility object, build a matrix of demand fle
     by (bus, hour) for this interval.
 """
 function _make_bus_demand_flexibility_amount(
-    case::Case, demand_flexibility::DemandFlexibility, start_index::Int, end_index::Int
+    case::Case,
+    demand_flexibility::DemandFlexibility,
+    start_index::Int,
+    end_index::Int,
+    bus_demand::Matrix,
+    sets::Sets,
 )::Tuple{Matrix,Matrix}
-    # Bus weighting
-    zone_to_bus_shares = _make_bus_demand_weighting(case)
-    # Demand flexibility up profile
-    simulation_demand_flex_amt_up = Matrix(
-        demand_flexibility.flex_amt_up[start_index:end_index, 2:end]
-    )
-    # Demand flexibility down profile
-    simulation_demand_flex_amt_dn = Matrix(
-        demand_flexibility.flex_amt_dn[start_index:end_index, 2:end]
-    )
+
+    # combine two profiles with user input overriding DOE flexibility
+    simulation_demand_flex_amt_up = zeros(end_index - start_index + 1, sets.num_bus)
+    simulation_demand_flex_amt_dn = zeros(end_index - start_index + 1, sets.num_bus)
+
+    # DOE flexibility, convert to MW, DOE timestamps already aligned
+    if demand_flexibility.enable_doe_flexibility
+        doe_flex_pct = Matrix(demand_flexibility.doe_flex_amt[start_index:end_index, 2:end])
+        doe_flex_mw = doe_flex_pct .* permutedims(bus_demand[sets.doe_flexible_bus_idx, :])
+        for i in 1:length(sets.doe_flexible_bus_idx)
+            simulation_demand_flex_amt_up[:, sets.doe_flexible_bus_idx[i]] = doe_flex_mw[
+                :, i
+            ]
+            simulation_demand_flex_amt_dn[:, sets.doe_flexible_bus_idx[i]] = doe_flex_mw[
+                :, i
+            ]
+        end
+    end
+
+    # replace columns with user input numbers if applicable
+    if !isnothing(demand_flexibility.flex_amt_up)
```

Comment:
At least for this iteration, yes. There should be some logic in `read.jl` that throws an error if `flex_amt_up` and `flex_amt_dn` aren't both specified. In future iterations, we may want different logic so that load shedding can be considered. Perhaps that would be a different input though.